src/js/components/DataTable/Body.js

```diff
@@ -39,127 +39,158 @@ const Row = memo(
-            pinnedOffset={pinnedOffset?._grommetDataTableSelect}
-            aria-disabled={isDisabled || !onSelect || undefined}
-            column={{
-              pin: Boolean(pinnedOffset?._grommetDataTableSelect),
-              plain: 'noPad',
-              size: 'auto',
-              render: () => (
-                <CheckBox
-                  tabIndex={onClickRow === 'select' ? -1 : undefined}
-                  a11yTitle={`${
-                    isSelected ? 'unselect' : 'select'
-                  } ${primaryValue}`}
-                  checked={isSelected}
-                  disabled={isDisabled || !onSelect}
-                  onChange={() => {
-                    if (isSelected) {
-                      onSelect(
-                        selected.filter((s) => s !== primaryValue),
-                        datum,
-                      );
-                    } else onSelect([...selected, primaryValue], datum);
-                  }}
-                  pad={cellProps.pad}
-                />
-              ),
-            }}
-            verticalAlign={verticalAlign}
-          />
-        )}
+              border={cellProps.pinned.border || cellProps.border}
+              pinnedOffset={pinnedOffset?._grommetDataTableSelect}
+              aria-disabled={isDisabled || !onSelect || undefined}
+              column={{
+                pin: Boolean(pinnedOffset?._grommetDataTableSelect),
+                plain: 'noPad',
+                size: 'auto',
+                render: () => (
+                  <CheckBox
+                    tabIndex={onClickRow === 'select' ? -1 : undefined}
+                    a11yTitle={`${
+                      isSelected ? 'unselect' : 'select'
+                    } ${primaryValue}`}
+                    checked={isSelected}
+                    disabled={isDisabled || !onSelect}
+                    onChange={() => {
+                      if (isSelected) {
+                        onSelect(
+                          selected.filter((s) => s !== primaryValue),
+                          datum,
+                        );
+                      } else onSelect([...selected, primaryValue], datum);
+                    }}
+                    pad={cellProps.pad}
+                  />
+                ),
+              }}
+              verticalAlign={verticalAlign}
+            />
+          )}
 
-        {rowDetails && (
-          <ExpanderCell
-            context={isRowExpanded ? 'groupHeader' : 'body'}
-            expanded={isRowExpanded}
-            onToggle={() => {
-              if (isRowExpanded) {
-                setRowExpand(rowExpand.filter((s) => s !== index));
-              } else {
-                setRowExpand([...rowExpand, index]);
+          {rowDetailsRender && (
+            <ExpanderCell
+              context={isExpanded ? 'groupHeader' : 'body'}
+              expanded={isExpanded}
+              disabled={!rowExpandable}
```

Comment:
```suggestion
              expandable={rowExpandable}
```
Can we change this prop name to `expandable`? I think the semantic html meaning of 'disabled' doesn't align with what we are doing here.